scipy/stats/tests/test_distributions.py

```diff
@@ -5780,11 +5780,24 @@ def test_argus_rvs_large_chi(self):
         x = stats.argus.rvs(50, size=500, random_state=325)
         assert_almost_equal(stats.argus(50).mean(), x.mean(), decimal=4)
 
-    def test_argus_rvs_ratio_uniforms(self):
-        # test that the ratio of uniforms algorithms works for chi > 2.611
-        x = stats.argus.rvs(3.5, size=1500, random_state=1535)
-        assert_almost_equal(stats.argus(3.5).mean(), x.mean(), decimal=3)
-        assert_almost_equal(stats.argus(3.5).std(), x.std(), decimal=3)
+    @pytest.mark.parametrize('chi, random_state', [
+            [0.1, 325],   # chi <= 0.5: rejection method case 1
+            [1.3, 155],   # 0.5 < chi <= 1.8: rejection method case 2
+            [3.5, 135]    # chi > 1.8: transform conditional Gamma distribution
+        ])
+    def test_rvs(self, chi, random_state):
+        x = stats.argus.rvs(chi, size=500, random_state=random_state)
+        _, p = stats.kstest(x, "argus", (chi, ))
+        assert_(p > 0.05)
+
+    @pytest.mark.parametrize('chi', [1e-9, 1e-6])
+    def test_rvs_small_chi(self, chi):
```

Comment:
Consider a more direct test e.g. https://github.com/scipy/scipy/pull/11954/files#r754718823

Reply:
personally I find the existing test easier to understand: in your example, you correctly "burn" the first 100 samples when reusing the seed. you can do that since the rejection constant is almost equal to one, but that might not be obvious. if ok for you, I would keep the existing test

Reply:
Sure, the existing test is a little easier to understand because you don't have to burn any samples. This test is just much stronger because it checks that the samples themselves are *correct* rather than that they "do not cause a null hypothesis to be rejected".